### PR TITLE
Refactor the Button into a function component

### LINKED ISSUE #4943: Refactor the Button into a function component.
### Expected Behavior

Hello Insomnia, 

I have some suggestions/improvements for the button component. 

1) For instance is there any way to click a disabled button? I noticed that there is a prop `onDisabledClick`, after a more deeply search I found that this prop never used so I guess that it is okay to remove it. 

2) Convert it to be a function component

### Actual Behavior

It's a Class component.

### Reproduction Steps

_No response_

### Is there an existing issue for this?

- [X] I have searched the [issue tracker](https://www.github.com/Kong/insomnia/issues) for this problem.

### Additional Information

_No response_

### Insomnia Version

2022.5.0-beta.2

### What operating system are you using?

macOS

### Operating System Version

macOS Monterey

### Installation method

download from insomnia.rest

### Last Known Working Insomnia version

_No response_

### PR DESCRIPTION
**Pull request by pavkout**
Highlights: 

* Converted the button to function component.
* Removed the onDisabledClick prop.
* Made some improvement.

Closes: #4943

### CODE REVIEW

File: packages/insomnia/src/ui/components/base/button.tsx [hunk 1]
```diff
@@ -17,34 +12,39 @@ export interface ButtonProps<T> {
...
+    children,
+    disabled,
+    tabIndex,
+    className,
+    type,
+    id,
+    style,
+    title,
+    onClick,
+  } = props;
```
Comment (dimitropoulos) [inline packages/insomnia/src/ui/components/base/button.tsx]: totally acceptable to destructure these inside the first argument (as long as you don't need the full props object anywhere, and this seems to be one of those places
```suggestion
export const Button = <T, >({
    value,
    disabled,
    tabIndex,
    className,
    type,
    id,
    style,
    title,
    onClick,
  }: PropsWithChildren<ButtonProps<T>>) => {
```

Also, this is a good place to use PropsWithChildren.  Check out `EnumSetting` or `NoticeRow` for examples.

Comment (pavkout) [reply packages/insomnia/src/ui/components/base/button.tsx]: @dimitropoulos You are right, many thanks :)

**Review by jackkav — APPROVED** [review]
Looks good I'm hesitant about the generic T personally. I like the approach illustrated below.
https://bobbyhadz.com/blog/react-typescript-extend-html-element-props
However the scope creep of this change is unnecessary in this pass.

**Comment by pavkout** [comment]
Hello @dimitropoulos and @jackkav,

I was thinking, how necessary is this button component. After some research I made I found that this button used in the following places. 

Following are all use cases: 

1. For the following components we gives `value` prop but never used the `onClick` prop (Also for Accessibility reason in this #4944 PR we have removed buttons from tabs)
- [ ] MarkdownEditor
- [ ] UnconnectedSettingsModal
- [ ] ResponsePane

2. For these we gives `value` and `onClick` prop, but never used the `value`.
- [ ] PasswordEditor
- [ ] AuthInputRow

3. It never used `value` prop
- [ ] GrpcTabbedMessages

4. For these we gives `value` and `onClick` prop and uses the `value` in `onClick` function
- [ ] AuthToggleRow
- [ ] Row (under key-value-editor folder)
- [ ] PromptModal
- [ ] RequestSwitcherModal
- [ ] WorkspaceEnvironmentsEditModal

By default the html button has the `value` attribute, we can use it directly. [link](https://www.w3schools.com/tags/att_button_value.asp)

Also the `MouseEvent` has the `value`, we can get it from there when `onClick` fires

My opinion is that it is safe to delete this component and use an html button instead. What do you think? What is your opinion?

If this works for you, I can make this change!

**Comment by jackkav** [comment]
> If this works for you, I can make this change!

Putting the value in the html is a nice idea, however the value attribute of `button` doesn't accept boolean, which is tricky to workaround. As you noticed its only in 5 places so not an insurmountable challenge. I'm just not sure which workaround is best. 🤷 

EDIT: @pavkout  It's probably better to address in a follow-up PR.

**Comment by pavkout** [comment]
@jackkav Oh man, I see, you are right. We have to see how to manage boolean/number values. I agree to address it in a follow-up PR.

**Review by marckong — APPROVED** [review]

**Review by filfreire — APPROVED** [review]
LGTM!

**Pull request merged**